### ISSUE TITLE
Operator count inconsistency with JuMP

### ISSUE THREAD

**Issue by jd-lara** [JuliaStochOpt/ParameterJuMP.jl#57]
When I run `sum(GAEv)` the operator counter in JuMP stays in 0 since it performs all the operations using `add_to_expression!`. However, if I run `sum(PAE)` the operator counter is different than 0. 

Some algebra method might be causing a call to a `+` operator instead of `add_to_expression`.

**Comment by jd-lara** [comment]
@joaquimg here is an example of the issue. 

```Julia
using JuMP
using Random
m = Model() 
I = 1:200
J = 1:200
vals = rand(length(I))
@variable(m, x[I, J] >= 0)
expression_container = JuMP.Containers.DenseAxisArray{JuMP.GenericAffExpr{Float64, VariableRef}}(undef, I)
for i in J
    expression_container[i] = sum(x[i,:]) - vals[i]
end
@constraint(m, con, sum(expression_container) >= 0);
m.operator_counter
0
```

```Julia
using ParameterJuMP
mwp = ModelWithParams()
I = 1:200
J = 1:200
vals = rand(length(I))
param_vals = add_parameters(mwp, vals)
@variable(mwp, x[I, J] >= 0)
expression_container = JuMP.Containers.DenseAxisArray{ParameterJuMP.DoubleGenericAffExpr{Float64, VariableRef, ParameterRef}}(undef, I)
for i in J
    expression_container[i] = sum(x[i,:]) - param_vals[i]
end
@constraint(mwp, con, sum(expression_container) >= 0);
mwp.operator_counter
548
```

For larger models, the way the expression algebra is implemented leads to the following warning in JuMP. 

```Julia
Warning: The addition operator has been used on JuMP expressions a large number of times. This warning is safe to ignore but may indicate that model generation is slower than necessary. For performance reasons, you should not add expressions in a loop. Instead of x += y, use add_to_expression!(x,y) to modify x in place. If y is a single variable, you may also use add_to_expression!(x, coef, y) for x += coef*y.
```

**Comment by jd-lara** [comment]
bump @joaquimg and @blegat about this. Any hints on how to address it?

**Comment by blegat** [comment]
In `sum(x[i,:]) - param_vals[i]` you call `Base.-` instead of `add_to_expression`, isn't that what's causing the counter to increase ?

**Comment by jd-lara** [comment]
I think if that were the case the counter would be other than 0 in the JuMP example

**Comment by jd-lara** [comment]
I solved it by calling `.destructive_add!` when adding PAE + variable.

**Comment by blegat** [comment]
So it was not in the sum ?

**Comment by jd-lara** [comment]
Yeah the sums of PAE are not adding counts.

**Comment by joaquimg** [comment]
You are calling destructive_add explicitly?
Where that is called in the example?
Doesn't looks like there is any PAE +
variable.
The problem is GAEv + param?

**Comment by jd-lara** [comment]
After some exploration, I realized what actually fixed the problem was the JuMP update to 0.20.0 and not the use of destructive_add, my guess is that it was this PR https://github.com/JuliaOpt/JuMP.jl/pull/1902. 

This is the test code and an example of the final implementation I did in PowerSimulations. 

```Julia
using ParameterJuMP
using JuMP
mwp = ModelWithParams()
I = 1:200
J = 1:200
vals = rand(length(I))
param_vals = add_parameters(mwp, vals)
@variable(mwp, x[I, J] >= 0)
expression_container = JuMP.Containers.DenseAxisArray{ParameterJuMP.DoubleGenericAffExpr{Float64, VariableRef, ParameterRef}}(undef, I)
for i in J
    expression_container[i] = sum(x[i,:])
    add_to_expression!(expression_container[i], - param_vals[i])
end
@constraint(mwp, con, sum(expression_container) >= 0);
mwp.operator_counter
```

In JuMP 0.19.2 and ParameterJuMP v0.1.1 this code results in 348 operations and in JuMP 0.20.0 and ParameterJuMP v0.1.2 results in 0. 

Sorry for the confusion, it wasn't clear what might have fixed the issue.
